.claude/agents/code-reviewer.md

```diff
@@ -0,0 +1,73 @@
+---
+name: code-reviewer
+description: Use this agent when you need expert code review after writing or modifying code. This agent should be called after completing any coding task to ensure quality, architectural compliance, and catch potential issues. Examples: <example>Context: The user has just implemented a new feature for processing SQLMesh snapshots. user: 'I just added a new method to handle snapshot fingerprinting in the Context class' assistant: 'Let me use the code-reviewer agent to analyze this implementation for potential issues and architectural compliance' <commentary>Since code was just written, use the code-reviewer agent to review the implementation for quality, edge cases, and adherence to SQLMesh patterns.</commentary></example> <example>Context: An agent just generated a database migration script. user: 'Here's the migration I created for adding a new state table' assistant: 'Now I'll have the code-reviewer agent examine this migration for safety and best practices' <commentary>Since a migration was created, use the code-reviewer agent to ensure it follows SQLMesh migration patterns and handles edge cases safely.</commentary></example>
+tools: Glob, Grep, LS, Read, NotebookRead, WebFetch, TodoWrite, WebSearch, Bash
+model: sonnet
```

Comment:
This comment is for all agents - should we not include tools/model and let the main determine what is allowed? I don't think we are ready to be opinionated on a detail like that for users.

Reply:
This is in the docs:
```
You have two options for configuring tools:

Omit the tools field to inherit all tools from the main thread (default), including MCP tools
Specify individual tools as a comma-separated list for more granular control (can be edited manually or via /agents)
```

I read that as the tools provided here override what I have already approved in my main thread.